src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -235,6 +251,93 @@ internal static void SetProperty(object resource, string propertyName, object va
+                }
+            }
+
+            foreach(ODataProperty property in oDataResource.Properties)
+            {
+                if(property.InstanceAnnotations != null)
+                {
+                    foreach (ODataInstanceAnnotation annotation in property.InstanceAnnotations)
+                    {
+                        AddInstanceAnnotationToContainer(instanceAnnotationContainer, deserializerProvider, readContext, annotation, property.Name);
+                    }
+                }
+            }
+        }
+
+        private static void AddInstanceAnnotationToContainer(IODataInstanceAnnotationContainer instanceAnnotationContainer, ODataDeserializerProvider deserializerProvider, 
+            ODataDeserializerContext readContext, ODataInstanceAnnotation annotation, string propertyName)
+        {
+            EdmTypeKind type;
+            IEdmTypeReference propertyType = null;
+
+            object annotationValue = ConvertValue(annotation.Value, ref propertyType, deserializerProvider, readContext, out type);
+
+            if (string.IsNullOrEmpty(propertyName))
+            {
+                instanceAnnotationContainer.AddAnnotation(annotation.Name, annotationValue);
+            }
+            else
+            {
+                instanceAnnotationContainer.AddPropertyAnnotation(propertyName,annotation.Name, annotationValue);
+            }
+        }
+
+        public static IODataInstanceAnnotationContainer GetAnnotationContainer(PropertyInfo propertyInfo, object resource)
+        {
+            IDelta delta = resource as IDelta;
+            object value;
+            if (delta != null)
+            {
+                delta.TryGetPropertyValue(propertyInfo.Name, out value);
+            }
+            else
+            {
+                value = propertyInfo.GetValue(resource);
+            }
+
+            IODataInstanceAnnotationContainer instanceAnnotationContainer = value as IODataInstanceAnnotationContainer;
+
+            if (instanceAnnotationContainer == null)
+            {
+                try
+                {
+                    if (propertyInfo.PropertyType == typeof(ODataInstanceAnnotationContainer) || propertyInfo.PropertyType == typeof(IODataInstanceAnnotationContainer))
+                    {
+                        instanceAnnotationContainer = new ODataInstanceAnnotationContainer();
+                    }
+                    else
+                    {
+                        instanceAnnotationContainer = Activator.CreateInstance(propertyInfo.PropertyType) as IODataInstanceAnnotationContainer;
+                    }
+
+                    if(delta != null)
+                    {
+                        delta.TrySetPropertyValue(propertyInfo.Name, instanceAnnotationContainer);
+                    }
+                    else
+                    {
+                        propertyInfo.SetValue(resource, instanceAnnotationContainer);
+                    }
+                }
```

Comment:
nit: generally add blank line after closing bracket (unless followed by else)

Reply:
Sorry; blank line should be before the catch, not here.  Sequential closing brackets generally do not have a blank line between (don't ask me why; I didn't write the guidelines... ;-))

---
In reply to: [488109607](https://github.com/OData/WebApi/pull/2219#discussion_r488109607) [](ancestors = 488109607)